plugins/module_utils/network/sonic/facts/bfd/bfd.py

```diff
@@ -231,7 +231,7 @@ def get_bfd_config(self, module):
             response = edit_config(module, to_request(module, request))
             if 'openconfig-bfd:bfd' in response[0][1]:
                 bfd_cfg = response[0][1].get('openconfig-bfd:bfd', None)
-        except ConnectionError as exc:
-            module.fail_json(msg=str(exc), code=exc.code)
+        except Exception as exc:
```

Comment:
Why is it better to return empty data than to fail with a message describing the failure as early as possible?

(It seems that if we "catch" the exception, something is likely to fail later with a harder to debug cause. Please explain if there is a reason you have in mind why it would be better to catch the exceptions here.)

Reply:
I see that this code has been restored to flag the exception if it occurs, so I am marking this issue "resolved".